### PR TITLE
Switch to Boto3.

### PR DESCRIPTION
**Pull request by b4hand**
Much of what s3po handles appears to now be core behavior in boto3, so that's why there's so much code removed. This is the replacement for #32.

### CODE REVIEW

**Comment by tony-bye** [comment]
Approved earlier version, but :+1:

File: s3po/backends/memory.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def download(self, bucket, key, fobj, retries, headers=None):
         else:
             fobj.write(obj)
 
-    def upload(self, bucket, key, fobj, retries, headers=None):
+    def upload(self, bucket, key, fobj, retries, headers=None, extra=None):
```
Comment (tony-bye) [inline s3po/backends/memory.py]: I thought `headers` were part of `extra` now?

Comment (tony-bye) [reply s3po/backends/memory.py]: @b4hand

Comment (b4hand) [reply s3po/backends/memory.py]: Yes, for S3. However, this interface is maintained as common for the Swift and "memory" backends. Unfortunately, with this change the headers aren't being transferred the same for Swift and S3, so code can't really be common. I could have written another compatibility layer, but ultimately our usage of Swift has essentially dropped off to nothing, and there's other differences between Swift and S3 that make it impossible to use the same code for both, so in reality, it was mostly a pipe dream to maintain compatibility between both Swift and S3.

**Pull request merged**